cmake/modules/SearchInstalledSoftware.cmake

```diff
@@ -36,7 +36,16 @@ if(NOT builtin_zlib)
    endif()
 endif()
 if(builtin_zlib)
+  set(CMAKE_ASM_OUTPUT_EXTENSION ".o")
+  set(CMAKE_ASM_OUTPUT_EXTENSION_REPLACE 1)
+  set(CMAKE_ASM_COMPILE_OBJECT       "<CMAKE_ASM_COMPILER> <FLAGS> -o <OBJECT> <SOURCE>")
+  set(CMAKE_ASM_SOURCE_FILE_EXTENSIONS s;asm;msa)
```

Comment:
Are these 4 lines really needed?

Reply:
@amadio, here in this zlib version we have one asm file to be compiled.. I was getting error:
```
CMake Error: Error required internal CMake variable not set, cmake may not be built correctly.
Missing variable is:
CMAKE_ASM_COMPILE_OBJECT
```
Mentioned lines by you had been fixed it for me.